test/unit/range/views/view_kmer_hash_test.cpp

```diff
@@ -17,155 +17,86 @@
+                                                std::list<seqan3::dna4> const,
+                                                std::forward_list<seqan3::dna4>,
+                                                std::forward_list<seqan3::dna4> const>;
+
+TYPED_TEST_SUITE(kmer_hash_properties_test, underlying_range_types, );
 
 class kmer_hash_test : public ::testing::Test
 {
 protected:
-    using result_t = std::vector<size_t>;
+    std::vector<seqan3::dna4> text1{"AAAAA"_dna4};
+    result_t result1{0,0,0}; // Same for gapped and ungapped
 
-    static constexpr auto ungapped_view = seqan3::views::kmer_hash(seqan3::ungapped{3});
-    static constexpr auto gapped_view = seqan3::views::kmer_hash(0b101_shape);
+    std::vector<seqan3::dna4> text2{"AC"_dna4};
+    result_t result2{}; // Same for gapped and ungapped
 
-    std::vector<seqan3::dna4> text1{"AAAAA"_dna4};
-    std::vector<seqan3::dna4> const ctext1{"AAAAA"_dna4};
-    result_t ungapped1{0,0,0};
-    result_t gapped1{0,0,0};
-
-    std::vector<seqan3::dna4> text2{"ACGTAGC"_dna4};
-    std::vector<seqan3::dna4> const ctext2{"ACGTAGC"_dna4};
-    result_t ungapped2{6,27,44,50,9};
-    result_t gapped2{2, 7, 8, 14, 1};
-
-    std::vector<seqan3::dna4> text3{"AC"_dna4};
-    std::vector<seqan3::dna4> const ctext3{"AC"_dna4};
-    result_t ungapped3{};
-    result_t gapped3{ungapped3};
-
-    seqan3::bitcompressed_vector<seqan3::dna4> text4{"ACGTAGC"_dna4};
-    seqan3::bitcompressed_vector<seqan3::dna4> const ctext4{"ACGTAGC"_dna4};
-    result_t ungapped4{ungapped2};
-    result_t gapped4{gapped2};
-
-    std::list<seqan3::dna4> text5{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4, 'A'_dna4, 'G'_dna4, 'C'_dna4};
-    std::list<seqan3::dna4> const ctext5{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4, 'A'_dna4, 'G'_dna4, 'C'_dna4};
-    result_t ungapped5{ungapped2};
-    result_t gapped5{gapped2};
-
-    std::forward_list<seqan3::dna4> text6{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4, 'A'_dna4, 'G'_dna4, 'C'_dna4};
-    std::forward_list<seqan3::dna4> const ctext6{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4, 'A'_dna4, 'G'_dna4, 'C'_dna4};
-    result_t ungapped6{ungapped2};
-    result_t gapped6{gapped2};
-
-    std::vector<seqan3::dna4> text7{"ACG"_dna4};
-    std::vector<seqan3::dna4> const ctext7{"ACG"_dna4};
-    result_t ungapped7{6};
-    result_t gapped7{2};
+    std::vector<seqan3::dna4> text3{"ACG"_dna4};
+    result_t ungapped3{6};
+    result_t gapped3{2};
 };
 
+TYPED_TEST(kmer_hash_properties_test, combinability)
+{
+    TypeParam text{'A'_dna4, 'C'_dna4, 'G'_dna4, 'T'_dna4, 'A'_dna4, 'G'_dna4, 'C'_dna4}; // ACGTAGC
+    result_t ungapped{6,27,44,50,9};
+    result_t gapped{2,7,8,14,1};
+    EXPECT_EQ(ungapped, text | ungapped_view | seqan3::views::to<result_t>);
+    EXPECT_EQ(gapped, text | gapped_view | seqan3::views::to<result_t>);
+
+    auto stop_at_t = seqan3::views::take_until([] (seqan3::dna4 x) { return x == 'T'_dna4; });
```

Comment:
```suggestion
    auto prefix_until_first_thymine = seqan3::views::take_until([] (seqan3::dna4 x) { return x == 'T'_dna4; });
```

this is confusing, `_t` normally means type